src/dackar/knowledge_graph/KGconstruction.py

```diff
@@ -0,0 +1,582 @@
+# Copyright 2024, Battelle Energy Alliance, LLC  ALL RIGHTS RESERVED
+"""
+Created on September, 2025
+
+@author: mandd, wangc
+"""
+# External Modules #
+import re
+import pandas as pd
+import os, sys
+import tomllib
+from jsonschema import validate, ValidationError
+import json
+import copy
+from pathlib import Path
+from datetime import datetime
+from dateutil.parser import parse
+from pandas.api.types import infer_dtype
+
+import logging
+
+currentDir = os.path.dirname(__file__)
+
+# Internal Modules #
+from dackar.knowledge_graph.py2neo import Py2Neo
+from dackar.knowledge_graph.visualize_schema import createIteractiveFile 
+from dackar.knowledge_graph.graph_utils import set_neo4j_import_folder
+from dackar.utils.mbse.customMBSEparser import customMBSEobject
+from dackar.utils.tagKeywordListReader import entityLibrary
+
+
+class KG:
+    """
+    Class designed to automate and check knowledge graph construction
+    """
+    def __init__(self, config_file_path, import_folder_path, uri, pwd, user):
+        """
+        Method designed to initialize the KG class
+        @ In, config_file_path, string, DBMS database folder
+        @ In, import_folder_path, string, folder which contains data to be imported
+        @ In, uri, string, uri = "bolt://localhost:7687" for a single instance or uri = "neo4j://localhost:7687" for a cluster
+        @ In, user, string, default to 'neo4j'
+        @ In, pwd, string, password the the neo4j DBMS database
+        @ Out, None
+        """
+        # Change import folder to user specific location
+        if import_folder_path is not None:
+            set_neo4j_import_folder(config_file_path, import_folder_path)
+
+        self.datatypes = ['string', 'integer', 'floating', 'boolean', 'datetime']
+
+        # Create python to neo4j driver
+        self.py2neo = Py2Neo(uri=uri, user=user, pwd=pwd)
+
+        self.graphSchemas = {} # dictionary containing the set of schemas of the knowledge graph
+
+        self.graphMetadata = {} # Metadata container of the knowledge graph --> TODO: discuss how to manage it
+
+        self.entityLibrary = entityLibrary(os.path.join(currentDir, os.pardir, os.pardir, os.pardir, 'data', 'tag_keywords_lists.xlsx'))
```

Comment:
We can use config file to provide the path to the specific data file. This can be changed in the future.